src/main/java/com/eternalcode/parcellockers/parcel/gui/ParcelListGUI.java

```diff
@@ -126,7 +133,7 @@ public List<String> replaceParcelPlaceholders(Parcel parcel, List<String> lore)
                 .toString()
             );
 
-        this.parcelLockerRepository.findByUUID(parcel.destinationLocker()).join().ifPresent(locker -> formatter
+        this.lockerRepository.findByUUID(parcel.destinationLocker()).join().ifPresent(locker -> formatter
```

Comment:
I suggest checking if it is in the cache first. Querying the database every time is a bit inappropriate when you have a cache.